tests/py_test.py

```diff
@@ -1278,3 +1279,77 @@ def test_job_timeout(timeout_db):
     table = table[0]
 
     assert table.committed() == False
+
+
+@pytest.fixture(scope='module')
+def sql_db(db):
+    with Postgresql() as postgresql:
+        conn = psycopg2.connect(**postgresql.dsn())
+        cur = conn.cursor()
+
+        cur.execute(
+            'CREATE TABLE test (id serial PRIMARY KEY, num integer, num2 integer, grp integer)'
+        )
+        cur.execute('INSERT INTO test (num, grp) VALUES (10, 0)')
+        cur.execute('INSERT INTO test (num, grp) VALUES (20, 0)')
+        cur.execute('INSERT INTO test (num, grp) VALUES (30, 1)')
+        conn.commit()
+
+        sql_params = postgresql.dsn()
+        sql_config = db.protobufs.SQLConfig(
+            hostaddr=sql_params['host'],
+            port=sql_params['port'],
+            dbname=sql_params['database'],
+            user=sql_params['user'],
+            adapter='postgres')
+
+        yield db, sql_config, cur
+
+        cur.close()
+        conn.close()
+
+
+@scannerpy.register_python_op(name='AddOne')
+def add_one(config, row: bytes) -> bytes:
+    row = json.loads(row.decode('utf-8'))
+    return json.dumps([{'id': r['id'], 'num2': r['num'] + 1} for r in row])
+
+
+def test_sql(sql_db):
+    (db, sql_config, cur) = sql_db
+
+    sql_query = db.protobufs.SQLQuery(
+        fields='test.id as id, test.num as num',
+        table='test',
+        id='test.id',
+        group='test.id')
+
+    row = db.sources.SQL(config=sql_config, query=sql_query)
```

Comment:
Could we document somewhere what the output format of `sources.SQL` is and the input format of `sink.SQL` is? As a comment is fine, but you could also put it here http://scanner.run/api/stdlib.html#built-in-sources